src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -1090,19 +1088,18 @@ bool emberAfTimeSynchronizationClusterSetUTCTimeCallback(
     const auto & timeSource  = commandData.timeSource;
 
     auto currentGranularity = TimeSynchronizationServer::Instance().GetGranularity();
-    if (granularity < GranularityEnum::kNoTimeGranularity || granularity > GranularityEnum::kMicrosecondsGranularity)
+    if (granularity == GranularityEnum::kUnknownEnumValue)
```

Comment:
is commandData.granularity guaranteed to have called `EnsureKnownEnumValue` to set things to unknown?

Reply:
same comment for below ... I would expect `EnsureKnownEnumValue` to be called whenever we validate against kUnknownEnumValue.

Reply:
How about this comment in the code?

```
    // All received enum values that are not listed above will be mapped
    // to kUnknownEnumValue. This is a helper enum value that should only
    // be used by code to process how it handles receiving and unknown
    // enum value. This specific should never be transmitted.
```

My understanding was that unknown enums that are received over the air are encoded to this value.

Reply:
> is commandData.granularity guaranteed to have called EnsureKnownEnumValue to set things to unknown?

Yes.